src/briefcase/commands/new.py

```diff
@@ -454,21 +452,21 @@ def build_app_context(self):
                 "GNU General Public License v3 (GPLv3)",
                 "GNU General Public License v3 or later (GPLv3+)",
                 "Proprietary",
-                "Other"
-            ],
```

Comment:
I should have reverted this along with reverting the refactor...although, I thought it would be innocuous to use a tuple here since I wouldn't expect anything to need to change this list of values.

Not that I'm pushing for a tuple, so mostly out of curiosity, why so adamant this is a list? (always looking to learn :) thanks)

Reply:
So: 
* A list is a structure of variable length, where all the items are of the same type, and there is no semantic significance to position other than ordering. Items can be arbitrarily added to a list.
* A tuple is of fixed length, items can (but don't have to be) be of different type, and there is a specific semantic interpretation of each position. Items cannot be arbitrarily added to a tuple; if something changes from an 3-tuple to a 4-tuple, the addition of the fourth element may change the interpretation of the first three.

Following that:
* "Here are your options, please choose" is a list, because each option is type equivalent, and changing the order doesn't change the interpretation of each item, just the index of the item.
* Size/Position are tuples, because a size is always (width, height), and position is (x,y) or (x,y,z).
* RGBA Color is a tuple; RGB are (usually) integers, A is a float, but the order isn't exchangeable; adding A to an RGB triple changes the interpretation of R, G, and B.
* A server `(host, port)` declaration is a tuple; host is a string, port is an integer.

As for why it matters - this is definitely in the weeds of "strong opinions". 

Functionally, the two will behave the same in most circumstances. However, there's an (admittedly weak) argument to be made about what you're communicating to the API user about the significance of position and type in the data you store in a list, vs a tuple. 

There's also a pedagogical concern. Python is a wonderful language that I love dearly, but one of its weaknesses from a teaching perspective is that the core data types are so powerful that new programmers often aren't exposed to the concept of data structures in any meaningful way. For beginning programmers (and, frankly, a large number of experienced programmers), this doesn't even matter because the core Python data types are so damn good - but one of the "big concepts" that you need to grok as you advance in skill is how data structures work, and how the choice of data structure can impact performance. Tuple vs List is a teaching moments - it's a chance to seed the idea that there are usually multiple ways to solve any problem, and choosing the right data structure is one of the things you need to do. It's very low stakes from a functional perspective, but it's also relatively easy to explain, and it provides a foundation for later discussions about why you might choose a list over a dictionary, or use something else entirely (like a linked list, or something else that isn't in the standard library).

The pedagogical concern is the bigger concern in my specific case here. I've spent a lot of energy encouraging contributors of all levels of experience; because of that, I sweat the small stuff like formatting and code structure so that the resulting codebase follows best practices.

Reply:
I see. That definitely makes sense. Thank you for your insight.